ocaml/xapi/xapi.ml

```diff
@@ -933,6 +934,7 @@ let server_init() =
           "Starting SR physical utilisation scanning", [Startup.OnThread], (Xapi_sr.physical_utilisation_thread ~__context);
           "Caching metadata VDIs created by foreign pools.", [ Startup.OnlyMaster; ], cache_metadata_vdis;
           "Stats reporting thread", [], Xapi_stats.start;
+          "Slave database backup", [ Startup.OnlySlave; Startup.OnThread; ], (Xapi_database_backup.endless_loop ~__context);
```

Comment:
This may be too late in the startup sequence: if any of the startup items above need the slave DB. We can do this immediately after establishing the master connection, so move it to the beginning of this list.